src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,130 @@
+        case "T":
+            ToDo newToDo = new ToDo(parsedInput[1], newTaskID);
+            this.taskList.add(newToDo);
+            return newToDo.toString();
+        case "D":
+            Deadline newDeadline = Deadline.createDeadline(parsedInput, newTaskID);
+            this.taskList.add(newDeadline);
+            return newDeadline.toString();
+        case "E":
+            Event newEvent = Event.createEvent(parsedInput, newTaskID);
+            this.taskList.add(newEvent);
+            return newEvent.toString();
+        default:
+            return "ugh how did we get here";
+        }
+    }
+    /**
+     * Marks a Task as complete and modifies the TaskList
+     *
+     * @param taskID Selected Task to mark as complete
+     *
+     * @return Description of the completed task
+     *
+     * @throws DukeException If it's an invalid task
+     */
+    public String completeTask(int taskID) throws DukeException {
+        verifyTaskValidity(taskID);
+        taskList.set(taskID - 1, taskList.get(taskID - 1).complete());
+        return taskList.get(taskID - 1).toString();
+    }
+    /**
+     * Deletes the specified Task and modifies the TaskList by decrementing the IDs of the remaining tasks
+     *
+     * @param taskID Selected Task to delete
+     *
+     * @return Description of the deleted task
+     *
+     * @throws DukeException If it's an invalid task
+     */
+    public String deleteTask(int taskID) throws DukeException {
+        currentTaskCount--;
+        verifyTaskValidity(taskID);
+        Task toDelete = taskList.get(taskID - 1);
+        taskList.remove(toDelete);
+        for (int i = taskID - 1; i < taskList.size(); i++) {
+            taskList.set(i, taskList.get(i).decrementID());
+        }
+        Task.decrementTaskCount();
+        return toDelete.toString();
+    }
+    /**
+     * Verifies the validity of the task to be handled, whether it exists in the TaskList or not
+     *
+     * @param taskID TaskID to be checked
+     *
+     * @throws DukeException Invalid task if it's more than the current list size or fewer than 1
+     */
+    private void verifyTaskValidity(int taskID) throws DukeException {
+        if (taskID > taskList.size()) {
+            throw new DukeException("invalid task: task id > list size");
+        } else if (taskID < 1) {
+            throw new DukeException("invalid task: task id < 1");
+        }
+    }
+    public ArrayList<Task> getAllTasks() {
+        return taskList;
+    }
+    /**
+     * Returns the String description of the remaining undone tasks in the list
+     *
+     * @return String description
+     */
+    public String getCurrentStatus() {
+        int incompleteTasks = 0;
+        for (Task t : taskList) {
+            if (!t.isComplete()) {
+                incompleteTasks++;
+            }
+        }
+        return "Now you have " + incompleteTasks + ((incompleteTasks == 1)
+                                                    ? " undone task"
```

Comment:
More indentations than should be? :)

Reply:
I honestly don't know what's to be done here. Doesn't the current indentation chosen make the ternary operator more readable? I'll KIV this and look into it before the final submission, thanks for pointing it out!